### PR TITLE
ci: also test against CentOS Stream 9

### PR DESCRIPTION
**Pull request by adrianreber**

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -125,3 +125,23 @@ jobs:
...
+
+  build_on_centos_stream_9:
+    runs-on: ubuntu-latest
+    name: Build on CentOS Stream 9
+    container:
+      image: quay.io/centos/centos:stream9
+    steps:
+    - name: Install git
+      run: dnf -y install git
+    - uses: actions/checkout@v2
```
Comment (martin-g) [inline .github/workflows/validate.yml]: v3 is [available](https://github.com/actions/checkout/tree/v3)
```suggestion
    - uses: actions/checkout@v3
```

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -125,3 +125,23 @@ jobs:
...
+      with:
+        fetch-depth: 0
```
Comment (martin-g) [inline .github/workflows/validate.yml]: Does it really need all history and all tags to be checked out ?
https://github.com/actions/checkout/blob/main/action.yml#L57

File: .github/workflows/validate.yml (hunk 1 above)
Comment (adrianreber) [reply .github/workflows/validate.yml]: changed

File: .github/workflows/validate.yml (hunk 2 above)
Comment (adrianreber) [reply .github/workflows/validate.yml]: not sure, probably just some historic copy and paste. also removed

**Pull request merged**